src/python/modules/ensembl/compara/plugins/TestFiles.py

```diff
@@ -0,0 +1,361 @@
+            self.ref_path = os.path.dirname(self.ref_path)
+        self.target_path = os.path.abspath(target_path)
+        self._target_tree_only = self._get_dir_cmp(self.target_path)
+        if os.path.isfile(self.target_path):
+            self.target_path = os.path.dirname(self.target_path)
+        self.common_tree = copy.deepcopy(self.ref_only)
+        # Recursive nested function (closure) to compare reference and target trees and build the common one
+        def cmp_trees(ref_node: Dict, target_node: Dict, common_node: Dict) -> None:
+            """Compares reference and target trees, moving their shared structure to the common tree."""
+            ref_dirnames = set(ref_node.keys())
+            target_dirnames = set(target_node.keys())
+            # Remove directories only found on the reference tree
+            for dirname in ref_dirnames.difference(target_dirnames):
+                del common_node[dirname]
+            for dirname in common_node.keys():
+                if dirname != '.':
+                    cmp_trees(ref_node[dirname], target_node[dirname], common_node[dirname])
+                else:
+                    # Remove files only found on the reference tree
+                    for filename in ref_node[dirname].difference(target_node[dirname]):
+                        common_node[dirname].remove(filename)
+                    # Remove common files from reference and target trees
+                    for filename in common_node[dirname]:
+                        ref_node[dirname].remove(filename)
+                        target_node[dirname].remove(filename)
+                # Remove emptied directories
+                if not common_node[dirname]:
+                    del common_node[dirname]
+                if not ref_node[dirname]:
+                    del ref_node[dirname]
+                if not target_node[dirname]:
+                    del target_node[dirname]
+        # Compare the directory trees from their root
+        cmp_trees(self._ref_tree_only, self._target_tree_only, self.common_tree)
+
+    @staticmethod
+    def _get_dir_cmp(path: str) -> Dict:
+        """Returns the directory tree rooted at the given path.
+
+        Note:
+            Files of each folder will be under the key ``.``.
+
+        Args:
+            path: Root path.
+
+        """
+        if os.path.isfile(path):
+            tree_dict = {'.': set(os.path.basename(path))}  # type: Dict
+        else:
+            tree_dict = {'': {}}
+            for dirpath, dirnames, filenames in os.walk(path):
+                dirpath = dirpath.replace(path, '')
+                subtree = reduce(operator.getitem, dirpath.split(os.sep), tree_dict)
+                subtree.update({name: {} for name in dirnames})
+                subtree['.'] = set(filenames)
+            tree_dict = tree_dict['']
+        return tree_dict
+
+    @staticmethod
+    def _prune_tree(root: Dict, paths: Union[str, List] = None, raise_err: bool = False) -> Dict:
```

Comment:
I reserve the right to suggest some changes to this function later, I'm not 100% sold yet :sunglasses: